src/near2far.cpp

```diff
@@ -465,6 +495,17 @@ dft_near2far fields::add_dft_near2far(const volume_list *where, double freq_min,
       default: abort("invalid normal direction in dft_near2far!");
     }
 
+    for (int i = 0; i < 2; ++i) {
+      if (has_direction(v.dim, fd[i]) &&
+          boundaries[High][fd[i]] == Periodic && boundaries[Low][fd[i]] == Periodic &&
+          float(w->v.in_direction(fd[i])) == float(v.in_direction(fd[i]))) {
```

Comment:
It seems the problem lies in this particular if statement, particularly the third and final test condition involving `in_direction`, which always evaluates to `false` and thus the subsequent initialization of the arrays `periodic_d`, `periodic_n`, `period`, and `periodic_k` to non-zero values is never performed.